pkg/cloud/vsphere/services/govmomi/create.go

```diff
@@ -256,34 +257,89 @@ func Create(ctx *context.MachineContext, bootstrapToken string) error {
 	}
 
 	userData64 := base64.StdEncoding.EncodeToString([]byte(userDataYAML))
+	var task *object.Task
+
+	builder, err := NewVmSpecBuilder(ctx)
+	if err != nil {
+		return err
+	}
 
-	// Use the appropriate path if we're connected to a vCenter
 	if ctx.Session.IsVC() {
-		return cloneVirtualMachineOnVCenter(ctx, userData64)
+		// Use the appropriate path if we're connected to a vCenter
+		spec, err := builder.BuildCloneSpec(userData64)
+		if err != nil {
+			return err
+		}
+		ctx.Logger.V(6).Info("vcenter cloning machine", "clone-spec", spec)
+		task, err = builder.Src.Clone(ctx, builder.Folder, ctx.Machine.Name, *spec)
+		if err != nil {
+			return errors.Wrapf(err, "error trigging clone op for machine %q", ctx)
+		}
+
+	} else {
+		// ESXi does not support cloning, we need to "ghetto" clone the VM
```

Comment:
Hi @moshloop,

Please use a different word rather than "ghetto".

Reply:
This is in reference to other ghetto tools e.g. https://github.com/lamw/ghettoVCB